src/index.js

```diff
@@ -49,17 +49,28 @@ class DBLAPI extends EventEmitter {
        * @param {error} error The error
        */
 
+      this.interval = null
+
       this.client = client;
-      this.client.on('ready', () => {
+      const setup = () => {
         this.postStats()
           .then(() => this.emit('posted'))
           .catch(e => this.emit('error', e));
-        setInterval(() => {
+
+        if (this.interval) clearInterval(this.interval)
+
+        this.interval = setInterval(() => {
           this.postStats()
             .then(() => this.emit('posted'))
             .catch(e => this.emit('error', e));
         }, this.options.statsInterval);
-      });
+      }
+
+      if (client.readyAt !== null || client.startTime !== null) {
```

Comment:
These variables do not accurately describe if the client's state is ready or not, which is why this wasn't originally implemented.

Reply:
Do you think maybe checking all the unavailable guilds to see if it's empty would be a proper way of checking? By the time they are all available the ready event would've already been sent

Reply:
Actually for discord.js `client.ws.status === 0` would tell if the client is ready.

I see a shard status on eris too, I'm looking into it now
(shard.status === 'ready')

Reply:
Do both d.js and eris have unavailable guilds stored on startup? Plus how would that react with bigger bots or sharding? Either way, seems like a hacky solution

Reply:
ws status is not the same as gateway status

Reply:
> ws status is not the same as gateway status

https://github.com/discordjs/discord.js/blob/stable/src/client/websocket/WebSocketManager.js#L425 shown here, the status is sent when the ready event is sent. Which is where it's listening. Ready in this case being when the guilds are all available not when the READY event is sent through the gateway.

Reply:
Also in eris client.ready is set at ready event

https://github.com/abalabahaha/eris/blob/master/lib/gateway/ShardManager.js#L87